integration_create_delete_test.go

```diff
@@ -22,43 +22,150 @@ const USER_DATA_TEMPLATE =
 set -e
 echo '%s' > "%s"
 `
-
 // An integration test that runs an EC2 instance, uses create_command to take a snapshot of it, and then delete_command
-// to delete that snapshot. Since testing these functions requires a lot of setup in an AWS account, it's faster and
-// easier to do a single integration test rather than a number of smaller unit tests that each do a lot of setup and
-// teardown.
+// to delete that snapshot.
 func TestCreateAndDelete(t *testing.T) {
 	t.Parallel()
 
-	logger := log.New(os.Stdout, "TestCreateAndDelete ", log.LstdFlags)
+	logger, ui := createLoggerAndUi("TestCreateAndDelete")
+	session := session.New(&aws.Config{Region: aws.String(AWS_REGION_FOR_TESTING)})
+	svc := ec2.New(session)
 
+	instance, instanceName := launchInstance(svc, logger, t)
+	defer terminateInstance(instance, svc, logger, t)
+	waitForInstanceToStart(instance, svc, logger, t)
+
+	snapshotId := takeSnapshotWithVerification(instanceName, *instance.InstanceId, ui, svc, logger, t)
+	deleteSnapshotWithVerification(instanceName, snapshotId, ui, svc, logger, t)
+}
+
+// An integration test that runs an EC2 instance, uses create_command to take a snapshot of it, and then calls the
+// delete_command to delete that snapshot, but setting the older than parmaeter in a way that should prevent any actual
+// deletion.
+func TestDeleteRespectsOlderThan(t *testing.T) {
+	t.Parallel()
+
+	logger, ui := createLoggerAndUi("TestDeleteRespectsOlderThan")
 	session := session.New(&aws.Config{Region: aws.String(AWS_REGION_FOR_TESTING)})
 	svc := ec2.New(session)
 
-	instance, uniqueId := launchInstance(svc, logger, t)
+	instance, instanceName := launchInstance(svc, logger, t)
 	defer terminateInstance(instance, svc, logger, t)
 	waitForInstanceToStart(instance, svc, logger, t)
 
-	ui := &cli.BasicUi{
-		Reader:      os.Stdin,
-		Writer:      os.Stdout,
-		ErrorWriter: os.Stderr,
-	}
+	snapshotId := takeSnapshotWithVerification(instanceName, *instance.InstanceId, ui, svc, logger, t)
+	// Always try to delete the snapshot at the end so the tests don't litter the AWS account with snapshots
+	defer deleteSnapshotWithVerification(instanceName, snapshotId, ui, svc, logger, t)
 
-	snapshotId := takeSnapshot(instance, uniqueId, svc, ui, logger, t)
-	waitForSnapshotToBeAvailable(snapshotId, svc, logger, t)
+	// Set olderThan to "10h" to ensure the snapshot, which is only a few seconds old, does not get deleted
+	deleteSnapshotForInstance(instanceName, "10h", 0, ui, logger, t)
+	waitForSnapshotToBeDeleted(snapshotId, svc, logger, t)
 	verifySnapshotWorks(snapshotId, svc, logger, t)
+}
+
```

Comment:
I was wondering how you'd handle testing deletion of AMIs older than X hours in an automated test. In addition to the nice date-parsing unit tests you wrote, maybe a test that validates that ec2-snapper parses the AMI's "create date" correctly and correctly computes the number of hours since it was created (which in the case of a test would be a few seconds) would be helpful.

But per a previous comment, I already have high confidence in that, so not too worried about it.